internal/cherrycheck.go

```
@@ -0,0 +1,254 @@
+	}
+
+	// Pass 2: patch-id comparison (handles multi-commit squash merges)
+	return patchIDCheck(repoPath, gitPath, upstream, branchRef, maxCommits)
+}
+
+// cherryCheck runs `git cherry` and returns true if all commits are applied.
+func cherryCheck(repoPath, gitPath, upstream, branchRef string) (bool, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, gitPath, "cherry", upstream, branchRef)
```

golangci-lint (2.11.4)

[medium] G204: Subprocess launched with variable

(gosec)